examples/prediction/category_prediction/webdata.py

```diff
@@ -0,0 +1,298 @@
+"""
+import os
+import random
+import sys
+import zipfile
+from operator import itemgetter
+
+import numpy as np
+import prettytable
+from prettytable import PrettyTable
+
+from nupic.frameworks.opf.model_factory import ModelFactory
+
+# List of page categories used in the dataset
+PAGE_CATEGORIES = [
+  "frontpage", "news", "tech", "local", "opinion", "on-air", "misc", "weather",
+  "msn-news", "health", "living", "business", "msn-sports", "sports", "summary",
+  "bbs", "travel"
+]
+
+# Configure the sensor/input region using the "SDRCategoryEncoder" to encode
+# the page category into SDRs suitable for processing directly by the TM
+SENSOR_PARAMS = {
+  "verbosity": 0,
+  "encoders": {
+    "page": {
+      "fieldname": "page",
+      "name": "page",
+      "type": "SDRCategoryEncoder",
+      # The output of this encoder will be passed directly to the TM region,
+      # therefore the number of bits should match TM's "inputWidth" parameter
+      "n": 1024,
+      # Use ~2% sparsity
+      "w": 21
+    },
+  },
+}
+
+# Configure the temporal memory to learn a sequence of page SDRs and make
+# predictions on the next page of the sequence.
+TM_PARAMS = {
+  "seed": 1960,
+  # Use "nupic.bindings.algorithms.TemporalMemoryCPP" algorithm
+  "temporalImp": "tm_cpp",
+  # Should match the encoder output
+  "inputWidth": 1024,
+  "columnCount": 1024,
+  # Use 1 cell per column for first order prediction.
+  # Use more cells per column for variable order predictions.
+  "cellsPerColumn": 1,
+}
+
+# Configure the output region with a classifier used to decode TM SDRs back
+# into pages
+CL_PARAMS = {
+  "implementation": "cpp",
+  "regionName": "SDRClassifierRegion",
+  # alpha parameter controls how fast the classifier learns/forgets. Higher
+  # values make it adapt faster and forget older patterns faster.
+  "alpha": 0.001,
+  "steps": 1,
+}
+
+# Create a simple HTM network that will receive the current page as input, pass
+# the encoded page SDR to the temporal memory to learn the sequences and
+# interpret the output SDRs from the temporary memory using the SDRClassifier
+# whose output will be a list of predicted next pages and their probabilities.
+#
+#   page => [encoder] => [TM] => [classifier] => prediction
+#
+MODEL_PARAMS = {
+  "version": 1,
+  "model": "HTMPrediction",
+  "modelParams": {
+    "inferenceType": "TemporalMultiStep",
+
+    "sensorParams": SENSOR_PARAMS,
+
+    # The purpose of the spatial pooler is to create a stable representation of
+    # the input SDRs. In our case the category encoder output is already a
+    # stable representation of the category therefore adding the spatial pooler
+    # to this network would not help and could potentially slow down the
+    # learning process
+    "spEnable": False,
+    "spParams": {},
+
+    "tmEnable": True,
+    "tmParams": TM_PARAMS,
+
+    "clParams": CL_PARAMS,
+  },
+}
+
+TRAINING_RECORDS = 100000
```

Comment:
I recommend taking this down to 10,000. When I ran with 10K I got:

```
 - Prediction Accuracy: 0.614173228346
 - Accuracy Predicting Top 3 Pages: 0.825196850394
```

This is close enough, I think, and it makes the example run so much faster. You might note that accuracy will increase if more records are used for learning. 

Also, instead of `TRAINING_RECORDS` I would change to this var name to something regarding learning. I don't like using the term "training" with HTM. Maybe `START_LEARNING_AFTER` or similar?